src/host/consoleInformation.cpp

```diff
@@ -179,6 +180,34 @@ void CONSOLE_INFORMATION::SetBracketedPasteMode(const bool enabled) noexcept
     _bracketedPasteMode = enabled;
 }
 
+void CONSOLE_INFORMATION::CopyTextToClipboard(const std::wstring_view text)
+{
+    const auto window = ServiceLocator::LocateConsoleWindow();
+    if (window)
+    {
+        // The clipboard can only be updated from the main GUI thread, so we
+        // need to post a message to trigger the actual copy operation. But if
+        // the pending clipboard content is already set, a message would have
+        // already been posted, so there's no need to post another one.
+        const auto clipboardMessageSent = _pendingClipboardText.has_value();
+        _pendingClipboardText = text;
+        if (!clipboardMessageSent)
+        {
+            PostMessageW(window->GetWindowHandle(), CM_UPDATE_CLIPBOARD, 0, 0);
+        }
+    }
+}
+
+std::optional<std::wstring> CONSOLE_INFORMATION::UsePendingClipboardText()
+{
+    // Once the pending text has been used, we clear the variable to let the
+    // CopyTextToClipboard method know that the last CM_UPDATE_CLIPBOARD message
+    // has been processed, and future updates will require another message.
+    const auto clipboardText = _pendingClipboardText;
+    _pendingClipboardText.reset();
+    return clipboardText;

```

Comment:
We can avoid a copy here by doing this:
```cpp
return std::exchange(_pendingClipboardText, {});
```

I've tested OSC 52 with Windows Terminal with up to 100MiB and it was relatively snappy. So I think avoiding a copy may be useful to reduce the peak memory usage.

...now that I mention this, I realize that immediately copying the clipboard contents into an `HGLOBAL` would have another benefit... Well, that's a minor issue though.